src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php

```diff
@@ -32,7 +33,21 @@ public function isFresh($timestamp)
             return false;
         }
 
-        return @filemtime($this->filePath) <= $timestamp;
+        // has the file *not* been modified? Definitely fresh
+        if (@filemtime($this->filePath) <= $timestamp) {
+            return true;
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($this->class);
+        } catch (\ReflectionException $e) {
+            // the class does not exist anymore!
+            return false;
+        }
+
+        $newSelf = AutowirePass::createResourceForClass($reflectionClass);
+
+        return $this == $newSelf;
```

Comment:
@nicolas-grekas I'm sorry, what would be the point of the `touch()` - I'm not sure I understand?

And yes, would could definitely cast into arrays to compare the properties only - that's definitely "in the spirit" of what we want to compare. Really, we could just compare the `autowiringMetadata` paths (as we've already checked things on `filePath`). But I didn't want to make this public just for this.

So:

* +1 for casting both objects into arrays
* I don't understand the `touch($this->filepath)` - would that just be to update the file so that it *looks* fresh automatically next time and avoids extra checking? If so, we could do that for sure - but probably the rest of the logic is very fast anyways
* I don't understand the comment `You're right, it's always false, hence the tests would pass anyway`.

Sorry for being slow on this :) - thanks to both of you for the help!